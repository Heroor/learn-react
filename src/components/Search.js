import React from 'react'
import PropTypes from 'prop-types'

const Search = ({
  value,
  onChange,
  onSubmit,
  children
}) =>
  <form onSubmit={onSubmit}>
    <input
      type="text"
      value={value}
      onChange={onChange}
    />
    <button type="submit">
      {children}
    </button>
  </form>


export default Search

Search.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.node
}

Search.defaultProps = {
  children: 'search...',
  value: ''
}