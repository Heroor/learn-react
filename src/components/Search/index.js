import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Button from '../Button'

class Search extends Component {
  componentDidMount() {
    this.input && this.input.focus()
  }
  render() {
    const {
      value,
      onChange,
      onSubmit,
      children
    } = this.props

    return (
      <form onSubmit={onSubmit}>
        <input
          type="text"
          ref={node => {this.input = node}}
          value={value}
          onChange={onChange}
        />
        <Button type="submit">
          {children}
        </Button>
      </form>
    )
  }
}

Search.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.node
}

Search.defaultProps = {
  children: 'search...',
  value: ''
}

export default Search
