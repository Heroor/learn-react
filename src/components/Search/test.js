import React from 'react'
import ReactDOM from 'react-dom'
import Search from './index'

import renderComponentToSnapshot from '../../config/TestConfig'

// Search test
describe('Search', () => {
  const props = {
    value: 'ipt val',
    onChange() {
      console.log('onChange')
    },
    onSubmit() {
      console.log('onSubmit')
    }
  }
  it('renders withou crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Search {...props}>{props.children}</Search>, div)
  })
  test('has a valid snapshot', () => {
    renderComponentToSnapshot(
      <Search {...props}>{props.children}</Search>
    )
  })
  test('has a valid text snapshot', () => {
    let text = 'Lorem ipsum dolor sit amet quisquam.'
    expect(text).toMatchSnapshot()
  })
})