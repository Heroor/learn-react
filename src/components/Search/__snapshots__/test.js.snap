// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search has a valid snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value="ipt val"
  />
  <button
    className=""
    onClick={undefined}
    type="submit"
  >
    search...
  </button>
</form>
`;

exports[`Search has a valid text snapshot 1`] = `"Lorem ipsum dolor sit amet quisquam."`;
